File: 7_kyu/I_Will_Take_the_Biggest_One_and_Nothing_Else.c
#include <inttypes.h>

// Returns maximal N = a * d with n = a + d and a != d.
// Returns -1 if n < 5
int64_t max_int_chain(const int64_t n) {
  if( n < 5 )
    return -1;
  const int64_t a = (n-1)/2;
  const int64_t b = n-a;
  return a*b;
}

File: 7_kyu/I_Will_Take_the_Biggest_One_and_Nothing_Else.py
# Returns maximal N = a * d with n = a + d and a != d.
# Returns -1 if n < 5
def max_int_chain(n: int) -> int:
  if n < 5:
    return -1
  a = (n-1)//2
  b = n-a
  return a*b
